stdlib/public/core/StringLegacy.swift

```diff
@@ -33,14 +33,25 @@ extension String {
       self = count == 0 ? "" : repeatedValue
       return
     }
+    guard !repeatedValue.isEmpty else {
+      self = ""
+      return
+    }
 
-    // TODO(String performance): We can directly call appendInPlace
-    var result = String()
-    result.reserveCapacity(repeatedValue._guts.count &* count)
-    for _ in 0..<count {
-      result += repeatedValue
+    var repeatedValue = repeatedValue
+    self = repeatedValue.withUTF8 { repeatedUTF8 in
+      String(_uninitializedCapacity: repeatedUTF8.count * count) { buffer in
```

Comment:
Paranoid question (not sure who can answer this):

Is `string.withUTF8 { $0.count }` _guaranteed_ to be equal to `string.utf8.count`? One could imagine a scenario where the underlying buffer could have some extra space allocated and even initialized. It seems like that would be a pitfall, so I hope Swift wouldn't do that, but do we have a documented guarantee somewhere?

Reply:
Yes, it has to be the same. `withUTF8` doesn't include the null terminator (nor does the UTF8View of course).